frontend/src/Leagues.tsx

```diff
@@ -0,0 +1,201 @@
+import React from "react";
+import Box from "@mui/joy/Box";
+import { extendedLeagues, leagueTiers, START_YEAR } from "./constants";
+import {
+    HEADER_HEIGHT_PX,
+    HEADER_MARGIN_PX,
+    TABLE_BTN_HEIGHT_PX,
+    TABLE_ELEMENTS_GAP,
+    BUTTON_SELECTOR_HEIGHT,
+} from "./styles/sizes";
+import { LeagueParams, LeagueStats, LeagueTier } from "./types";
+import { getLeagueLabel, range, toPercent } from "./utils";
+import ButtonSelector from "./ButtonSelector";
+import Table, { ColHeaderData, RowData } from "./Table";
+import TableLayout from "./TableLayout";
+
+const TABLE_TOP_POSITION =
+    HEADER_HEIGHT_PX +
+    HEADER_MARGIN_PX +
+    BUTTON_SELECTOR_HEIGHT * 2 +
+    TABLE_BTN_HEIGHT_PX +
+    TABLE_ELEMENTS_GAP * 4;
+
+interface Props {
+    params: LeagueParams;
+    setParams: React.Dispatch<React.SetStateAction<LeagueParams>>;
+    stats: LeagueStats;
+    loading: boolean;
+    error: string | null;
+    refresh: () => void;
+}
+
+export default function Leagues({
+    params,
+    setParams,
+    stats,
+    loading,
+    error,
+    refresh,
+}: Props) {
+    const availableYears = range(START_YEAR, new Date().getFullYear() + 1);
+
+    return (
+        <Box>
+            <ButtonSelector
+                current={params.year}
+                options={availableYears.reverse()}
+                setCurrent={(year) =>
+                    setParams((params) => ({ ...params, year }))
+                }
+                style={{ marginBottom: `${TABLE_ELEMENTS_GAP}px` }}
+            />
+
+            <Box
+                display="flex"
+                alignItems="center"
+                justifyContent="center"
+                overflow="auto"
+            >
+                <Box display="flex" alignItems="center" mr={1} fontSize="sm">
+                    Base:
+                    <ButtonSelector
+                        current={params.tier}
+                        options={leagueTiers.slice()}
+                        setCurrent={(tier) =>
+                            setParams((params) => ({
+                                ...params,
+                                league: "GeneralLeague",
+                                tier,
+                            }))
+                        }
+                        getLabel={getLeagueTierLabel}
+                        variant="outlined"
+                        style={{ marginLeft: "5px" }}
+                    />
+                </Box>
+
+                <Box display="flex" alignItems="center" fontSize="sm">
+                    Extended:
+                    <ButtonSelector
```

Comment:
This is what I landed on for the visual separation of base leagues - is "extended" the right term to use for other leagues? And how do you feel about the UI choice?

![image](https://github.com/user-attachments/assets/08b74377-0c76-4ffc-9f8a-b95bc51e760e)

Reply:
I think it's great for now, and we can be open to feedback if anyone wants changes.
*Maybe* we could consider a splash of color for fun? Or font diffs? There's a lot of room to play there, but for MVP this is fantastic.

I would change `Extended` to `Expansions` my 0.02$

Reply:
`Expansions`! That was the word I wanted, oops 😂 

A splash of color would be nice, will make a non-MVP issue for adding some polish to this!